### PR TITLE
fix: Update dependency, fix skin rendering, fix redis connection,

### PR DESCRIPTION
**Pull request by k-nero**

### CODE REVIEW

**Review by DuckySoLucky — APPROVED** [review]
lgtm, didn't test

not sure if that's still the issue but before latest version of canvas and sharp had some conflicts and they wouldn't install correctly, an error would occur, can you confirm that has been fixed?

File: public/resources/ts/stats-defer.ts [hunk 1]
```diff
@@ -79,18 +76,19 @@ if (calculated.skin_data) {
...
-
-      // Arm swing
-      const armRotation = Math.cos(time) * 0.03 + basicArmRotationZ;
-      player.skin.leftArm.rotation.z = armRotation;
-      player.skin.rightArm.rotation.z = armRotation * -1;
-
-      // Cape wave
-      player.cape.rotation.x = Math.sin(time) * 0.01 + basicCapeRotationX;
-    });
+    // skinViewer.animation.add((player, time) => {
```
Comment (DuckySoLucky) [inline public/resources/ts/stats-defer.ts]: also what's the point of this code? Either remove it or uncomment :shrug:

File: src/app.js [hunk 2]
```diff
@@ -495,10 +495,10 @@ app.all("/manifest.webmanifest", async (req, res) => {
   }));
   res.json(Object.assign({ shortcuts }, manifest));
 });
-
-app.all("/:player/:profile?", async (req, res, next) => {
-  res.redirect(`/stats${req.path}`);
-});
+//
```
Comment (DuckySoLucky) [inline src/app.js]: same with this

**Review by DuckySoLucky — APPROVED** [review]

File: public/resources/ts/stats-defer.ts (hunk 1 above)
Comment (k-nero) [reply public/resources/ts/stats-defer.ts]: it's no longer work in the new version of skinview3d, so i commented it

File: src/app.js (hunk 2 above)
Comment (k-nero) [reply src/app.js]: this one may lead to some unexpected redirecting, so comment it too

File: public/resources/ts/stats-defer.ts (hunk 1 above)
Comment (DuckySoLucky) [reply public/resources/ts/stats-defer.ts]: can you just remove it then? there's no point in having it then

**Comment by k-nero** [comment]
The skin can now display normally, but the cape is not fixed yet, still under investigation

File: src/app.js (hunk 2 above)
Comment (DuckySoLucky) [reply src/app.js]: never had any kind of redirection issues but sure idk why that even exists, feel free to remove that too

File: public/resources/ts/stats-defer.ts (hunk 1 above)
Comment (k-nero) [reply public/resources/ts/stats-defer.ts]: sure

File: src/redis.js
Comment (k-nero) [inline src/redis.js:1]: "credentials.json" also needs some changes to avoid connection errors

Comment (DuckySoLucky) [reply src/redis.js:1]: seems like a windows issue, never had any issues on arch linux. you can change that at https://github.com/SkyCryptWebsite/SkyCrypt/blob/development/src/credentials.js#L14

**Comment by k-nero** [comment]
> The skin can now display normally, but the cape is not fixed yet, still under investigation

**Comment by k-nero** [comment]
> > The skin can now display normally, but the cape is not fixed yet, still under investigation

@DuckySoLucky Cape rendering is now working properly, The previous issue may have come from api.ashcon.app/mojang/v2/user

**Review by DuckySoLucky — CHANGES_REQUESTED** [review]
~~updating deps seems to have broken prettier (tested on dev branch and it works completely fine)~~

Solved

**Comment by DuckySoLucky** [comment]
Merge https://github.com/k-nero/SkyCrypt/pull/2 (fix for the prettier)

**Comment by DuckySoLucky** [comment]
(we love prettier updates)

**Comment by k-nero** [comment]
> Merge [k-nero#2](https://github.com/k-nero/SkyCrypt/pull/2) (fix for the prettier)

Thanks

**Review by DuckySoLucky — APPROVED** [review]
lgtm, tested

**Comment by DuckySoLucky** [comment]
~~@k-nero can you add eslint ignore comment to to the `/public/resources/ts/elements/guild-button.ts` (it seems like something broke, not entirely sure what or why)~~

https://github.com/k-nero/SkyCrypt/pull/3

**Comment by DuckySoLucky** [comment]
@k-nero prettier :kek:

**Comment by DuckySoLucky** [comment]
this was way more painful than it needed to be

**Comment by k-nero** [comment]
@DuckySoLucky yes, 
finally merged

**Pull request merged**